src/key.rs

```diff
@@ -26,6 +31,147 @@ pub enum Integer {
+            (Self::F64(lhs), Self::F64(rhs)) => TotalOrd(lhs) == TotalOrd(rhs),
+            _ => false,
+        }
+    }
+}
+
+impl Eq for OrderedFloat {}
+
+impl PartialOrd for OrderedFloat {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for OrderedFloat {
+    fn cmp(&self, other: &Self) -> Ordering {
+        use ordered_float::OrderedFloat as TotalOrd;
+        match (*self, *other) {
+            (Self::F32(_), Self::F64(_)) => Ordering::Less,
+            (Self::F64(_), Self::F32(_)) => Ordering::Greater,
+            (Self::F32(lhs), Self::F32(rhs)) => TotalOrd(lhs).cmp(&TotalOrd(rhs)),
+            (Self::F64(lhs), Self::F64(rhs)) => TotalOrd(lhs).cmp(&TotalOrd(rhs)),
+        }
+    }
+}
+
+impl Hash for OrderedFloat {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        use ordered_float::OrderedFloat as TotalOrd;
+        match *self {
+            Self::F32(v) => TotalOrd(v).hash(state),
+            Self::F64(v) => TotalOrd(v).hash(state),
+        }
+    }
+}
+
+impl TryFrom<f32> for OrderedFloat {
+    type Error = Error;
+
+    fn try_from(v: f32) -> Result<Self, Self::Error> {
+        Ok(Self::F32(v))
+    }
+}
+
+impl TryFrom<f64> for OrderedFloat {
+    type Error = Error;
+
+    fn try_from(v: f64) -> Result<Self, Self::Error> {
+        Ok(Self::F64(v))
+    }
+}
+
+impl From<OrderedFloat> for FloatProxy {
+    fn from(ordered: OrderedFloat) -> Self {
+        match ordered {
+            OrderedFloat::F32(v) => FloatProxy::F32(v),
+            OrderedFloat::F64(v) => FloatProxy::F64(v),
+        }
+    }
+}
+
+/// A float serialization policy which rejects any attempt to serialize a float with an error.
+#[derive(Debug, Clone, Ord, PartialOrd, Eq, PartialEq, Hash, Serialize, Deserialize)]
+pub enum RejectFloat {}
+
+impl TryFrom<f32> for RejectFloat {
+    type Error = Error;
+    fn try_from(_: f32) -> Result<Self, Self::Error> {
+        Err(Error::UnsupportedType("f32"))
+    }
+}
+
+impl TryFrom<f64> for RejectFloat {
+    type Error = Error;
+    fn try_from(_: f64) -> Result<Self, Self::Error> {
+        Err(Error::UnsupportedType("f64"))
+    }
+}
+
+impl From<RejectFloat> for FloatProxy {
+    fn from(this: RejectFloat) -> Self {
+        match this {}
+    }
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum FloatProxy {
+    F32(f32),
+    F64(f64),
+}
+
+/// A policy for handling floating point types in a `Key`.
+///
+/// Currently there are two important `FloatPolicy` types: [`RejectFloat`] and
+/// [`OrderedFloat`]. The former will emit errors instead of allowing floats to
+/// be serialized and the latter while serialize them and provide a total order
+/// which does not adhere to the IEEE standard.
+///
+/// [`RejectFloat`]: RejectFloat
+/// [`OrderedFloat`]: OrderedFloat
+pub trait FloatPolicy:
+    TryFrom<f32, Error = Error>
+    + TryFrom<f64, Error = Error>
+    + Into<FloatProxy>
```

Comment:
I wasn't sure about adding `visit_float` and `deserialize_f32/f64` since it would mean exposing foreign types/traits through Serde; though I suppose this is already explicitly a serde-compatible crate, so I guess that's okay?

Reply:
Yeah. That's not an issue. Thanks!